middleware/src/util/semver/semver.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2018 Shift Devices AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file is copied from the bitbox-wallet-app (https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/util/semver/semver.go).
+// Modifications:
+// - change errp to errors.New()
```

Comment:
I copy and pasted the semver.go packet from the App. I'm not to happy with it, but I think it's better to `vendor/` bigger parts of the App into the Base repository. Happy for suggestions here as well (especially from @benma).